components/mpas-ocean/src/shared/mpas_ocn_tracer_DMS.F

```diff
@@ -374,8 +374,8 @@ subroutine ocn_tracer_DMS_surface_flux_compute(activeTracers, DMSTracers, forcin
          DMS_forcing%surfacePressure(column) = atmosphericPressure(iCell)*PascalsToAtmospheres
          DMS_forcing%iceFraction(column) = iceFraction(iCell)
 !maltrud assume for now that if there is any land ice, it is all land ice
-         if (associated(landIceFraction)) then
-            if (landIceFraction(iCell) > 0.0_RKIND) DMS_forcing%iceFraction(column) = 1.0_RKIND
+         if (associated(landIceFloatingFraction)) then
+            if (landIceFloatingFraction(iCell) > 0.0_RKIND) DMS_forcing%iceFraction(column) = 1.0_RKIND
          endif
```

Comment:
This is an exotic case but I think should use `landIceMask` (rather than `landIceFraction` or `landIceFloatingFraction`.
```
          if (associated(landIceMask)) then
            if (landIceMask(iCell) == 1) DMS_forcing%iceFraction(column) = 1.0_RKIND
```
It seems like we want DMS to think it's under ice regardless of the type (sea ice or grounded or floating land ice).  Then again, it presumably doesn't matter in the thin film region either way.

Reply:
No problem. Commit dropped.